gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java

```diff
@@ -161,18 +168,19 @@ public void setTaskDuration(long duration) {
    * Update record-level metrics.
    *
    * @param recordsWritten number of records written by the writer
+   * @param branchIndex fork branch index
    */
-  public void updateRecordMetrics(long recordsWritten) {
+  public void updateRecordMetrics(long recordsWritten, int branchIndex) {
     JobMetrics metrics = JobMetrics.get(this.getProp(ConfigurationKeys.JOB_NAME_KEY), this.jobId);
+    String forkBranchId = ForkOperatorUtils.getForkId(this.taskId, branchIndex);
 
```

Comment:
Ok, make more sense. But then wouldn't you have to do the same thing for the Job-Level Records Metric?